docs/components/ar-hit-test.md

```diff
@@ -0,0 +1,48 @@
+---
+title: hit-test
+type: components
+layout: docs
+parent_section: components
+source_code: src/components/hit-test.js
+examples: []
+---
+
+[hit-test]: https://immersive-web.github.io/hit-test/
+
+This component uses the WebXR hit-test API to position virtual objects in the real world. Remember to request the `hit-test` optional feature to allow it work.
```

Comment:
Can we perhaps automate this? The `ar-hit-test` component adds `hit-test` to WebXR/optionalFeatures?

Reply:
Yes I was thinking that but no other components do it.

If we do this then I think also:
* `webxr` should add `DOM-Overlay` if required
* `background` should add lighting-estimation
* this should also add `anchors` as well as `hit-test`

is that okay to do in this PR?

Reply:
Yeah I think it should work. the `init` method of `ar-hit-test` handles the setup

Reply:
What is the best way to do this?

```
var features = new Set(this.el.systems.webxr.data.optionalFeatures);
features.add('hit-test');
features.add('anchors');
this.el.setAttribute('webxr', 'optionalFeatures', Array.from())
```

Doesn't work, I think Systems don't handle the 3 argument setAttribute like components do.

IS it okay to update `this.el.systems.webxr.data.optionalFeatures` directly?

Reply:
I don't love much the API we have know (array typed properties are rare) but you can do.

```
  var optionalFeatures = sceneEl.getAttribute('webxr').optionalFeatures;
  optionalFeatures.push('hit-test');
  sceneEl.setAttribute('webxr', optionalFeatures);
```

Reply:
And that won't set the Overlay element and required features to null right?

Reply:
It was setting the OverlayElement to null so I did the following to ensure all the settings were preserved.

```javascript
    // Update WebXR to support light-estimation
    var webxrData = this.el.getAttribute('webxr');
    var optionalFeaturesArray = webxrData.optionalFeatures;
    if (!optionalFeaturesArray.includes('light-estimation')) {
      optionalFeaturesArray.push('light-estimation');
      this.el.setAttribute('webxr', webxrData);
    }
```